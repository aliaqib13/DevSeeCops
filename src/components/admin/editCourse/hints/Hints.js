import React, { Component } from 'react';
import {
    Button, Table, Modal, message,
} from 'antd';
import CreateHintModal from './create-hint-modal';
import EditHintModal from './edit-hint-modal';
import './hints.scss';
import HTMLSanitize from '../../../HTMLSanitize/HTMLSanitize';

const { Column } = Table;
const confirmModal = Modal.confirm;

class Hints extends Component {
    constructor(props) {
        super(props);
        this.state = {
            createModalVisible: false,
            editModalVisible: false,
            maxLength: 600,
        };
    }

    deleteHint = id => {
        confirmModal({
            title: 'Are You Sure to delete this hint?',
            okText: 'Yes',
            okType: 'primary',
            cancelText: 'No',
            onOk: () => {
                this.props.deleteHint(id).then(res => {
                    if (res === true) {
                        message.success('Deleted!');
                    } else {
                        message.error(res.message);
                    }
                });
            },
        });
    }

    toggleCreateModal = () => {
        this.setState({
            createModalVisible: !this.state.createModalVisible,
        });
    }

    toggleEditModal = hint => {
        this.setState({
            editModalVisible: !this.state.editModalVisible,
            editedHint: hint,
        });
    }

    render() {
        const {
            hints, addHint, course_id, editHint, toggleHintModal, visible,
        } = this.props;
        const {
            createModalVisible, editModalVisible, editedHint, maxLength,
        } = this.state;
        return (
            <Modal
                onCancel={toggleHintModal}
                title="Add Hint Message"
                visible={visible}
                footer={null}
                width={1000}
            >
                <div className="hints-container">
                    <div className="hints-list">
                        <Button className='add-hint' onClick={this.toggleCreateModal}> Add Hint</Button>
                        {
                            hints
                                ? (
                                    <Table
                                        dataSource={hints}
                                        rowKey={item => item.id}
                                        pagination={{ position: 'bottom', pageSize: 10 }}
                                    >
                                        <Column
                                            title="Hint Message "
                                            render={(text, record) => (
                                                <div>
                                                    <p><HTMLSanitize content={record.message.length > maxLength ? `${(record.message).substring(0, maxLength - 3)}...` : record.message} /></p>
                                                    )
                                                </div>
                                            )}
                                            key="message"
                                        />
                                        <Column
                                            title="Actions"
                                            key="actions"
                                            width={200}
                                            render={(text, record) => (
                                                <div className="actions-btns">
                                                    <Button
                                                        type="primary"
                                                        onClick={() => this.toggleEditModal(record)}
                                                    >
                                                        Edit
                                                    </Button>
                                                    <Button
                                                        type="danger"
                                                        style={{ marginLeft: '10px' }}
                                                        onClick={() => this.deleteHint(record.id)}
                                                    >
                                                        Delete
                                                    </Button>
                                                </div>
                                            )}
                                        />
                                    </Table>
                                )
                                : <div>There is no any hint items</div>
                        }
                        {
                            createModalVisible
                            && (
                                <CreateHintModal
                                    key="create-hint-modal"
                                    visible={createModalVisible}
                                    toggleModal={this.toggleCreateModal}
                                    addHint={addHint}
                                    course_id={course_id}
                                />
                            )
                        }

                        {
                            editModalVisible
                            && (
                                <EditHintModal
                                    visible={editModalVisible}
                                    toggleModal={this.toggleEditModal}
                                    hint={editedHint}
                                    editHint={editHint}
                                />
                            )
                        }

                    </div>
                </div>
            </Modal>
        );
    }
}

export default Hints;
