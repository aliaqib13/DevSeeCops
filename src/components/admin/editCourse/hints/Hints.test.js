import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Hints from './Hints';

describe('Hints', () => {
    let component;

    beforeEach(() => {
        component = shallow(<Hints />);
        component.setProps({ hints: [{ id: 5, course_id: 1, message: '<p>Test Hint</p>' }] });
    });

    it('Should render successfully', () => {
        const wrapper = component.find('.hints-container');
        expect(wrapper.exists()).toBeTruthy();
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should show button for adding hints', () => {
        const button = component.find('.add-hint');
        expect(button.exists()).toBeTruthy();
    });

    it('Should open modal when Add Hint button clicked', () => {
        const button = component.find('.add-hint');
        button.simulate('click');
        expect(component.state().createModalVisible).toEqual(true);
    });

    it('Should render hints in the table', () => {
        expect(component.find('withStore(Table)').props().dataSource).toEqual([{ id: 5, course_id: 1, message: '<p>Test Hint</p>' }]);
    });

    it('Should open edit modal when Edit button clicked', () => {
        component.find('withStore(Table)').props().children[1].props.render('test', { message: 'test', id: 1 }).props.children[0].props.onClick();
        expect(component.state().editModalVisible).toEqual(true);
    });
});
