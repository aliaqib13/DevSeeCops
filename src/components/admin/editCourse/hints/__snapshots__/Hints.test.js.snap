// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hints Should render successfully 1`] = `
<div
  className="hints-container"
>
  <div
    className="hints-list"
  >
    <Button
      block={false}
      className="add-hint"
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
    >
       Add Hint
    </Button>
    <withStore(Table)
      dataSource={
        Array [
          Object {
            "course_id": 1,
            "id": 5,
            "message": "<p>Test Hint</p>",
          },
        ]
      }
      pagination={
        Object {
          "pageSize": 10,
          "position": "bottom",
        }
      }
      rowKey={[Function]}
    >
      <Column
        key="message"
        render={[Function]}
        title="Hint Message "
      />
      <Column
        key="actions"
        render={[Function]}
        title="Actions"
        width={200}
      />
    </withStore(Table)>
  </div>
</div>
`;
